src/drops.js

```diff
@@ -46,21 +65,34 @@ export class Drop {
 
 		this.hex.drop = undefined;
 
-		if (this.health) {
-			creature.heal(this.health);
-			game.log('%CreatureName' + creature.id + '% gains ' + this.health + ' health');
+		if (this.alterations.health) {
+			creature.heal(this.alterations.health);
 		}
 
-		if (this.energy) {
-			creature.energy += this.energy;
-			game.log('%CreatureName' + creature.id + '% gains ' + this.energy + ' energy');
+		if (this.alterations.energy) {
+			creature.recharge(this.alterations.energy);
 		}
+
+		if (this.alterations.endurance) {
+			creature.restoreEndurance(this.alterations.endurance);
+		}
+
+		if (this.alterations.movement) {
+			creature.restoreMovement(this.alterations.movement);
+		}
+
+		// Log all the gained alterations.
+		const gainedMessage = Object.keys(this.alterations)
+			.map((key) => `${this.alterations[key]} ${key}`)
+			.join(', ')
+			// Replace last comma with "and".
+			.replace(/, ([^,]*)$/, ', and $1');
+		game.log(`%CreatureName${creature.id}% gains ${gainedMessage}`);
+
 		creature.player.score.push({
 			type: 'pickupDrop',
 		});
 
-		creature.updateAlteration(); // Will cap the stats
```

Comment:
No longer needed as the `creature.recharge()`, `creature.restoreMovement()` etc already cap the stats.